### PR TITLE
fix: update to properly initialize when brought into view

### LINKED ISSUE #5113: [BUG] sometimes playground tool json editor doesn't show the full text
**Describe the bug**
Sometimes the json tool editor only shows the first two lies of text when first rendered

**To Reproduce**
1. Go to playground
2. Click on add a tool
3. Scroll down 
4. See tool being cutoff

**Expected behavior**
The full tool should render in the editor

**Screenshots**
image

**Environment (please complete the following information):**
 - OS: mac
 - locally running phoenix

### PR DESCRIPTION
**Pull request by Parker-Stafford**
resolves #5113 

- fixes an issue where our json editors (and probably other code mirror editors) were being cut off or not rendering at all when initially rendered out of the view port
- checked datasets json editors for regressions

### CODE REVIEW

File: app/src/components/code/JSONEditor.tsx [hunk 1]
```diff
@@ -28,32 +46,79 @@ export type JSONEditorProps = Omit<
...
-      value={props.value}
-      extensions={extensions}
-      editable
-      theme={codeMirrorTheme}
-      {...props}
-    />
+    <div
+      ref={containerRef}
+      css={css`
+        min-height: ${!isInitialized ? JSON_EDITOR_MIN_HEIGHT : "auto"};
```
Comment (Parker-Stafford) [inline app/src/components/code/JSONEditor.tsx:112]: this is to make it so this container fits the size of the editor after it's been initialized

File: app/src/pages/playground/PlaygroundTool.tsx [hunk 2]
```diff
@@ -65,7 +65,6 @@ export function PlaygroundTool({
       backgroundColor={"yellow-100"}
       borderColor={"yellow-700"}
       variant="compact"
-      key={tool.id}
```
Comment (Parker-Stafford) [inline app/src/pages/playground/PlaygroundTool.tsx:68]: this key is on the outside element

File: app/src/components/generative/ToolChoiceSelector.tsx [hunk 3]
```diff
@@ -77,24 +77,26 @@ export function ToolChoicePicker({
       }}
     >
       {[
-        <Item key="auto">
+        <Item key="auto" textValue="auto">
```
Comment (Parker-Stafford) [inline app/src/components/generative/ToolChoiceSelector.tsx:80]: fixes some console warnings since the items children are not strings

**Comment by cephalization** [comment]
As discussed, I'm cool with this as long as we move the logic into a wrapping component that we can reuse and manage the complexity of separately

**Review by cephalization — APPROVED** [review]

**Pull request merged**